File: SelectionSort.c
/*
    Author : Md. Abid Hasan Roni
    Dept of CSE
    Faridpur Engineering College
*/


// Selection Sort
#include <stdio.h>

int MinLoc(int A[], int ib , int size){
    int i,loc;
    loc = ib;
    for(i=ib+1; i<size; i++){
        if(A[loc]>A[i])
            loc = i;
    }
    return loc;
}

void SelectionSort(int A[],int size){
    int i, loc ,temp;
    for(int i=0;i<size-1;i++){
        loc = MinLoc(A,i,size);
        if(loc!=i){
            temp = A[loc];
            A[loc] = A[i];
            A[i] = temp;
        }
    }
}

int main(){
    int size ,i ;
    printf("Enter the size of array : ");
    scanf("%d",&size);
    int A[size];
    printf("Enter Array element : ");
    for ( i = 0; i < size; i++){
        scanf("%d",&A[i]);
    }
    printf("Your entered array elements is : \n");
    for ( i = 0; i < size; i++){
        printf("%d ",A[i]);
    }   
    SelectionSort(A,size);
    printf("\nSelection Sorted array elements is : \n");
    for ( i = 0; i < size; i++){
        printf("%d ",A[i]);
    }       
    printf("\n");
    return 0;
}

File: SelectionSort.java
/* 
    Author : Md. Abid Hasan Roni
    Dept of CSE
    Faridpur Engineering College
*/

//Selection Sort

import java.util.*;

public class SelectionSort{

    public static void main(String[] args) {
        Scanner in=new Scanner(System.in);
        System.out.println("Enter Array Size : ");
        int i,n=in.nextInt();
        int[] A=new int[n];
        for(i=0;i<n;i++)
            A[i]=in.nextInt();
        System.out.println("Selection Sorted Value : ");
        Sort obj=new Sort();
        obj.Selection(A, n);
        for(i=0;i<n;i++)
            System.out.print(" "+A[i]);
        System.out.println();
    }
    
}

class Sort{
    int MinLoc(int A[],int lb,int n)
    {
        int i,loc=lb;
        for(i=lb+1;i<=n-1;i++)
        {
            if(A[loc]>A[i])
            {
                loc=i;
            }
        }
        return loc;
    }
    
    void Selection(int A[],int n)
    {
        int i ,loc,temp;
        for(i=0;i<n-1;i++)
        {
            loc=MinLoc(A,i,n);
            if(loc!=i)
            {
                temp=A[loc];
                A[loc]=A[i];
                A[i]=temp;
            }
        }
    }    
}